docs/how-to/uc-image-creation.rst

```diff
@@ -0,0 +1,307 @@
+
+.. note::
+
+    This guide assumes access to features available exclusively to `dedicated Snap Store`_ users.
+
+Canonical builds and publishes Ubuntu Core images for a range of `supported platforms`_.
+However, these images include the generic Linux kernel. 
+In order to run Ubuntu Core with the real-time kernel, we need to build it ourselves. 
+
+This guide shows how to build an Ubuntu Core image with the real-time kernel.
+We start by showing how to :ref:`build a vanilla image <ubuntu-core-image>` which is excellent for testing and tuning, and then illustrate how to :ref:`make a custom image <custom-ubuntu-core-image>` with production-ready kernel configurations.
+
+
+.. _ubuntu-core-image:
+
+Create an image with the real-time kernel
+-----------------------------------------
+
+To do this we need to describe the image content and then use a tool to build it.
+
+.. _model-assertion:
+
+Create the model assertion
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The model assertion is a digitally signed document that describes the content of the Ubuntu Core image.
+Read the `model assertion`_ documentation before continuing.
+
+Below is an example model assertion in YAML, describing a `core22` Ubuntu Core
+image:
+
+.. literalinclude :: uc-image-creation/model.json
+   :language: json
+
+Inside an empty directory, create a file named ``model.json`` with the above content.
+
+Change the following:
+
+- ``authority-id``, ``brand-id`` to your developer ID, since this is custom model. Use ``snapcraft whoami`` command to get your developer ID.
+- ``timestamp`` to an RFC3339 formatted string that falls within the validity period of your signing key. Use ``date -Iseconds --utc`` command to generate the current time.
+- ``store`` to your dedicated Snap Store ID.
+- ``model`` to a name that accurately represents your device(s).
+
+The ``snaps`` array is a list of snaps that get included in the image.
+In that list, the ``realtime-kernel`` snap contains the realtime Linux kernel.
+Here you can add any other snaps, including for example your real-time applications.
+
+Next, we need to sign the model assertion.
+Refer to the guide on `signing model assertion`_ for details on how to sign the model assertion. 
+
+Here are the needed steps:
+
+1) Create and register a key
+
+Use ``snapcraft list-keys`` to check your existing keys.
+If you don't already have a key, create one locally and register it with your account.
+Choose a meaningful name for the key, as it can be used to sign multiple models.
+In this guide we use ``rtu-model`` to sign all our real-time Ubuntu models.
+
+.. code-block:: shell
+
+    snapcraft create-key rtu-model
+    snapcraft register-key rtu-model
+
+Remember to update the model assertion's ``timestamp``, if you created a new key and plan to use it next.
+
+2) Sign the model assertion
+
+.. code-block:: shell
+
+    snap sign -k rtu-model model.json > model.signed.yaml
+
+The ``snap sign`` command takes JSON as input and produces YAML as output!
```

Comment:
```suggestion
The ``snap sign`` command takes JSON as input and produces YAML as output.
```

Sorry, is there a reason for ending the sentence with an exclamation point?

Reply:
This was because the tool taken a JSON, signs it and outputs YAML, which is unexpected. The conversion itself is surprising and YAML is actually a less portable serialization format, so I don't know why this is done.

Reply:
Ah ok. But i think even then we can stick with a "." instead of "!" in a how-to guide.